extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -35,15 +67,88 @@ export abstract class ToolBase implements ITool {
 		return this._statusDescription;
 	}
 
-	public loadInformation(): Promise<void> {
+	protected async getPip3InstallLocation(packageName: string): Promise<string> {
+		const command = `pip3 show ${packageName}`;
+		const pip3ShowOutput: string = (await this._platformService.runCommand(command));
+		const installLocation = /^Location\: (.*)$/gim.exec(pip3ShowOutput);
+		let retValue = installLocation && installLocation[1];
+		if (retValue === undefined || retValue === null) {
+			this.logToOutputChannel(`   >${command}`);
+			this.logToOutputChannel(`   Could not find 'Location' in the output:`);
+			this.logToOutputChannel(pip3ShowOutput, `   output:`);
+			return '';
+		} else {
+			return retValue;
+		}
+	}
+
+	protected logToOutputChannel(data: string | Buffer, header?: string): void {
+		this._platformService.logToOutputChannel(data, header);
+	}
+
+	public async install(): Promise<void> {
+		await this.installCore();
```

Comment:
What if this fails? try, catch?

Reply:
await throws if this fails right? if I add try catch I will just be throwing whatever i cath. The installCore() impmententaion already logs all errors encountered during executiton so I thik we are good here.